doc/api/single-executable-applications.md

```diff
@@ -0,0 +1,104 @@
+executes its contents, otherwise it operates like plain Node.js.
+
+A bundled JavaScript file can be turned into a single executable application
+with any tool which can inject resources into the `node` binary.
+
+One such tool is [postject][]:
+
+```console
+$ cat hello.js
+console.log(`Hello, ${process.argv[2]}!`);
+$ cp $(command -v node) hello
+# On systems other than macOS:
+$ npx postject hello NODE_JS_CODE hello.js \
+    --sentinel-fuse NODE_JS_FUSE_fce680ab2cc467b6e072b8b5df1996b2
+# On macOS:
+$ npx postject hello NODE_JS_CODE hello.js \
+    --sentinel-fuse NODE_JS_FUSE_fce680ab2cc467b6e072b8b5df1996b2 \
+    --macho-segment-name NODE_JS
+$ ./hello world
+Hello, world!
+```
+
+This currently only supports running a single embedded [CommonJS][] file.
+
+## Notes
+
+### `require(id)` in the injected module is not file based
+
+`require()` in the injected module is not the same as the [`require()`][]
+available to modules that are not injected. This also does not have any of the
+properties that non-injected [`require()`][] has except [`require.main`][]. It
+can only be used to load built-in modules. Attempting to load a module that can
+only be found in the file system will throw an error.
+
+Instead of relying on a file based `require()`, users can bundle their
+application into a standalone JavaScript file to inject into the executable.
+This also ensures a more deterministic dependency graph.
+
+However, if a file based `require()` is still needed, that can also be achieved:
+
+```js
+const { createRequire } = require('node:module');
+require = createRequire(__filename);
+```
+
+### `__filename` and `module.filename` in the injected module
+
+The values of `__filename` and `module.filename` in the injected module are
+equal to [`process.execPath`][].
+
+### `__dirname` in the injected module
+
+The value of `__dirname` in the injected module is equal to the directory name
+of [`process.execPath`][].
+
+### Single executable application creation process
+
+A tool aiming to create a single executable Node.js application is supposed to
+inject the contents of a JavaScript file into:
+
+* a resource named `NODE_JS_CODE` if the `node` binary is a [PE][] file
+* a section named `NODE_JS_CODE` in the `NODE_JS` segment if the `node` binary
+  is a [Mach-O][] file
+* a note named `NODE_JS_CODE` if the `node` binary is an [ELF][] file
+
+The tool should also search the binary for the
+`NODE_JS_FUSE_fce680ab2cc467b6e072b8b5df1996b2:0` [fuse][] string and flip the
+last character to `1` to indicate that a resource has been injected.
+
+### Linux support
+
+AMD64 Ubuntu is the only Linux distribution where single-executable support is
+tested regularly on CI currently, due to lack of better tools to generate
+single-executables that can be used to test this feature on other platforms.
+
+Suggestions for other resource injection tools/workflows are welcomed, please
+start a discussion at <https://github.com/nodejs/single-executable/discussions>
```

Comment:
Does this render properly in the page by the way?

Reply:
Yea, it looks like this:
<img width="1148" alt="Screenshot 2023-02-10 at 7 31 05 PM" src="https://user-images.githubusercontent.com/42526976/218110555-e8519731-a56c-45ea-915b-b3b0133c3553.png">